src/System.Net.Security/tests/FunctionalTests/NegotiateStreamTestForUnix.cs

```diff
@@ -453,6 +475,289 @@ public void NegotiateStream_StreamToStream_KerberosAuthInvalidTarget_Failure()
             }
         }
 
+        public static IEnumerable<object[]> ValidNtlmCredentials()
+        {
+         
+                yield return new object[]{new NetworkCredential(TestConfiguration.NtlmUser, _ntlmPassword, TestConfiguration.Domain)};
+                yield return new object[] {new NetworkCredential(TestConfiguration.NtlmUser, _ntlmPassword)};
+                yield return new object[]
+                {
+                    new NetworkCredential(
+                        string.Format(@"{0}\{1}", TestConfiguration.Domain, TestConfiguration.NtlmUser), _ntlmPassword)
+                };
+                yield return new object[]
+                {
+                    new NetworkCredential(
+                        string.Format("{0}@{1}", TestConfiguration.NtlmUser, TestConfiguration.Domain), _ntlmPassword)
+                };
+        }
+
+        public static IEnumerable<object[]> InvalidNtlmCredentials
+        {
+            get
+            {
+                yield return new object[] { new NetworkCredential(TestConfiguration.NtlmUser, _ntlmPassword, TestConfiguration.Domain.Substring(1))};
+                yield return new object[] { new NetworkCredential(TestConfiguration.NtlmUser.Substring(1), _ntlmPassword, TestConfiguration.Domain) };
+                yield return new object[] { new NetworkCredential(TestConfiguration.NtlmUser, _ntlmPassword.Substring(1), TestConfiguration.Domain) };
+                yield return new object[] { new NetworkCredential(string.Format(@"{0}\{1}", TestConfiguration.Domain, TestConfiguration.NtlmUser), _ntlmPassword, TestConfiguration.Domain.Substring(1))};
+                yield return new object[] { new NetworkCredential(string.Format("{0}@{1}", TestConfiguration.NtlmUser, TestConfiguration.Domain.Substring(1)), _ntlmPassword)};
+                yield return new object[] { new NetworkCredential(string.Format(@"{0}\{1}", TestConfiguration.Domain.Substring(1), TestConfiguration.NtlmUser), _ntlmPassword, TestConfiguration.Domain)};
+                yield return new object[] { new NetworkCredential(TestConfiguration.NtlmUser, _ntlmPassword, TestConfiguration.Realm)};
+
+            }
+        }
+
+
+        [Theory, OuterLoop]
+        [MemberData("ValidNtlmCredentials")]
```

Comment:
Master will need to be merged into dev/negotiatestream first (since there are merge conflicts), and this can be cleaned up in the branch before going back to master.